### DISCUSSION TITLE
confused in layout of MMA, need help

### DISCUSSION THREAD

**Discussion by liujshi** [NVIDIA/cutlass#1954] [Q&A]
![20241120115910](https://github.com/user-attachments/assets/c8eab303-6966-40f7-91c4-b5f6b0dc0a55)
i know the layout of
thr_mma.partition_A(a); is (MMA, MMA_M, MMA_K)
but what is layout of MMA, i got stucked.

as in picture above,  partition_A(a) gets layout of MMA is (_2, _2)
but  partition_B(b) is _2. can't understand.

can you help me? or show me some clues.

thanks.

**Comment by liujshi** [comment]
![image](https://github.com/user-attachments/assets/2b9a0f1f-58ea-42b1-bd9d-a22f122eb7af)
![image](https://github.com/user-attachments/assets/2490e706-e461-41c4-bb23-30d5a1466e94)

why the ALayout is different

**Reply by thakkarV** [reply]
That's just how the hardware works

**Reply by liujshi** [reply]
thank you， what the mean of the [ABC]Layout in MMA_traits? @thakkarV

**Reply by thakkarV** [reply]
Please read our documentation for mma atoms
